### PR TITLE
Remote updates

### PR DESCRIPTION
**Pull request by wearhere**
This will reduce in scope once the previous PRs have been merged.

Heads-up @rissem, Mac apps must be code-signed for Electron's auto-updater to
work, so we must disable the ability for the server to build the app (since only Macs can code-sign and it is unlikely that the server is a Mac).

So, no point bothering with the `electronDownload` helper anymore.

I don't think that this detracts from the library's utility overly much though since if the user still wants their webserver to serve the build they can just copy it into the `public` directory and it will work just like the old static serving:

```
// From project directory
cp .meteor-electron/final/app-darwin.zip public/app-darwin.zip

// App is now available at http://localhost:3000/app-darwin.zip
```

### CODE REVIEW

**Comment by rissem** [comment]
Just starting to look at this now, but my gut feeling is it's still a good thing for a lot of people if the app can be automatically built by the server (Windows builds, Mac builds not going through the app store, etc.)

File: README.md [hunk 1]
```diff
@@ -9,16 +9,38 @@ Meteor-electron strives to be the easiest way to create a desktop Meteor applica
...
 ## Configuration
 Limited configuration is possible via `Meteor.settings.electron` For example
 ```json
 {
   "electron": {
+    "name": "MyApp",
+    "icon": {
+      "osx": "build/MyApp.icns"
+    },
+    // Must conform to Semver: https://docs.npmjs.com/getting-started/semantic-versioning.
```
Comment (rissem) [inline README.md:28]: Nice to have these comments, but we should keep this as something that the user can copy/paste and is valid JSON. Maybe just say the same thing above the JSON code block?

File: README.md [hunk 2]
```diff
@@ -34,15 +56,10 @@ if (! Electron.isDesktop){
 }
 ```
 
-## Download Button
-
-Add a download button to your app with `{{> electronDownload}}`. This button points to `/download/darwin-x64`, `/download/linux-ia64`, etc. based on platform. Feel free to point to these locations manually yourself as well.
-
-The download button will not be shown within the Electron application.
-
 ## Deploy
```
Comment (rissem) [inline README.md:60]: Not really comfortable with this yet.

**Comment by wearhere** [comment]
@rissem the app needs to be code-signed even if it will be distributed _outside_ the MAS, unfortunately—it's just a requirement of Electron's auto-updater.

**Comment by wearhere** [comment]
Looking ahead to Windows, [we'll need to install Wine](https://github.com/maxogden/electron-packager#building-windows-apps-from-non-windows-platforms) which will probably be a lot easier locally if it's even possible on the server (the Meteor package would have to install a binary/non-NPM dependency, for whatever server architecture, using who knows what permissions).

Also, Windows [needs to be code-signed](https://github.com/atom/grunt-electron-installer#sign-your-installer-or-else-bad-things-will-happen) too and even if an arbitrary server can do that, it still wouldn't be turnkey—the user would need to provide their certificate.

So, basically my feeling is that `meteor-electron`:
- _can_ and _should_ (and does, currently) support turn-key packaging for local use, but
- packaging for remote use _will_ need a bit of configuration, and be done locally—and that's ok given that it's not _that_ much configuration and not that hard to copy the locally-built file to a server or even just the public folder.

File: README.md (hunk 1 above)
Comment (wearhere) [inline README.md:28]: Good idea. Let's tackle that when the README as a whole is polished for distribution: https://github.com/rissem/meteor-electron/issues/9#issuecomment-161782284

File: README.md [hunk 3]
```diff
@@ -9,7 +9,12 @@ Meteor-electron strives to be the easiest way to create a desktop Meteor applica
 
 `meteor add quark:electron`
 
-After adding the package, an electron app wrapper will be while you develop.
+After adding the package, an electron app wrapper will be built and opened.
+
+This app, as well as the ready-to-distribute binaries (see [Deploy](#deploy)), is built within
+`YOUR_PROJECT_DIRECTORY/.meteor-electron`. This allows the apps to be easily located as well as
+(forthcoming) the apps to be cached for speedier startup. You should add this directory to your
+`.gitignore`.
```
Comment (rissem) [inline README.md:17]: Not sure if it's better to do this in a tmp directory (that will occasionally be deleted and cause extra builds) or a new directory within the project that requires the user to update their `.gitignore`

File: README.md (hunk 2 above)
Comment (rissem) [inline README.md:60]: Might still be nice to have helpers and a standard place to put the apps even if we're not building on production.

File: README.md (hunk 3 above)
Comment (wearhere) [inline README.md:17]: The problem isn't that the tmp directory might be deleted so much as that it's unpredictable. That means that the builds can't be found after the fact, e.g. by external scripts that publish the builds.

I do think it's reasonable to let the user configure where the build directory is located. So then the only question is, what's the default? My feeling was that a known directory was better because basically everyone is going to want the directory to be known, so that they can grab the builds—and maybe some people will not want to use `.meteor-electron`, but a bunch of people will be cool with it.

**Comment by wearhere** [comment]
Did you manually merge this PR @rissem? If so, let's close this out and continue the threads above in new issues.
